starknet_py/contract.py

```diff
@@ -646,10 +642,11 @@ def compute_address(
         compiled = create_compiled_contract(
             compilation_source, compiled_contract, search_paths
         )
+        assert compiled.abi is not None
```

Comment:
I still don't like the fact that `create_compiled_contract` returns an Optional abi. I don't think there's ever the case that ABI is None when building `ContractClass` from the source code. 

Having `assert compiled.abi is not None` in multiple places of the library is the proof that this is not an ergonomic approach.

Reply:
I fully agree. However the asserts are used only after `create_compiled_contract`, which is already **deprecated**.
Introducing a subclass (probably with its schema as well) and extra logic would just increase complexity of a function that is already going to be removed.

Reply:
Does the same problem exist if using `create_contract_class`? Because it's here to stay I think, and if abi is typed to be `Optional` the same issue will exist even when compiling contracts with external tool.

Reply:
https://github.com/software-mansion/starknet.py/pull/708